test/functional/test_framework/wallet.py

```diff
@@ -55,25 +55,34 @@ def get_utxo(self, *, txid=''):
             index = self._utxos.index(utxo)
         return self._utxos.pop(index)
 
-    def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_spend=None):
+    def send_self_transfer(self, *, fee_rate=Decimal("0.003"), seq_num = 0xffffffff, from_node, utxo_to_spend=None, send_tx = True, default_fee = False):
```

Comment:
It seems like it might be misleading to have a method whose name is `send_self_transfer`, but then have a flag to say `send = False`. 

Instead, maybe it would be more straightforward to have a separate method `create_self_transfer`, that returns the tx hex. This function could be used within `send_self_transfer` to create the transaction before broadcasting it. 

I'm about to submit a PR for another test where I had a go at doing it this way.

Reply:
That totally makes sense! or maybe we can simply rename the method to something that mimics it's accurate behavior, something in the lines of `sign_or_send_self_transfer`. Would love to head @MarcoFalke's thoughts on this